src/dft/backends/rocfft/commit.cpp

```diff
@@ -0,0 +1,426 @@
+/*******************************************************************************
+* Copyright Codeplay Software Ltd
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include <array>
+#include <algorithm>
+#include <optional>
+
+#include <rocfft.h>
+#include <hip/hip_runtime_api.h>
+
+#include "oneapi/mkl/dft/detail/commit_impl.hpp"
+#include "oneapi/mkl/dft/detail/descriptor_impl.hpp"
+#include "oneapi/mkl/dft/detail/types_impl.hpp"
+#include "oneapi/mkl/dft/types.hpp"
+#include "oneapi/mkl/exceptions.hpp"
+#include "rocfft_handle.hpp"
+
+namespace oneapi::mkl::dft::rocfft {
+namespace detail {
+
+// rocfft has global setup and cleanup functions which use some global state internally.
+// Each can be called multiple times in an application, but due to the global nature, they always need to alternate.
```

Comment:
by alternate you mean the clean_plan() before any recommit right ?

Reply:
No, this paragraph is about `rocfft` as a library itself. I mean each `rocfft_setup` needs to be paired with a `rocfft_cleanup` before `rocfft_setup` should be called again and vice versa.